src/colorizer/utils/react_utils.ts

```diff
@@ -318,6 +319,17 @@ export type AnnotationState =  {
   setVisibility: (visible: boolean) => void;
   selectionMode: AnnotationSelectionMode;
   setSelectionMode: (mode: AnnotationSelectionMode) => void;
+  /**
+   * For a given ID, returns the range of IDs that would be selected if the ID
+   * is clicked with range selection mode turned on.
+   * @param dataset The dataset to look up track information from.
+   * @param id The ID of an object to select with range rules applied.
+   * @returns One of the following:
+   * - Returns `null` if no range would be selected.
+   * - A list of IDs to select if this ID is clicked with range selection mode turned on.
+   */
+  getSelectRangeFromId: (dataset: Dataset, id: number) => number[] | null;
+  handleAnnotationClick: (dataset: Dataset, id: number, selectRange?: boolean) => void;
```

Comment:
Nit: Because this is a "State" class, (actually just a type? any reason why it's not a class? I'm not really sure what this thing is... in fact maybe it should be in its own file...), it might be nice to have the actual data elements of the object (like currentLabelIx, selectionMode,...) collected together and separate from the function methods.

Reply:
It's just a type. It extends the actual `AnnotationData` class which holds the core annotation-related data, which knows nothing about the UI. The react hook here (`useAnnotationState`) wraps the methods of `AnnotationData` with UI-related methods and data.

Reply:
I would like to move `useAnnotationState` into its own file at some point though... I think there's a TODO for that